### importing necessary libraries

In [1]:
import os
import pandas as pd
import time

from pathlib import Path # for system paths related tasks
import inspect # to get information about live objects like modules, classes, functions, methods
import psutil as ps
import sweetviz as sv

import warnings
warnings.filterwarnings("ignore")

### loaing installed python packages

In [2]:
# imoprting report directory for saving reports files.
from src.utils.config import REPORTS

Lahore Real Estate Price Prediction Pipeline


In [3]:
# get Root directory of this note book
ROOT_DIR = Path.cwd().parent

In [4]:
# importing data ingestion module for data loading.
from scripts.data import data_ingestion

### Downloading & Loading Lahore House Listings from Zameen.com

In [5]:
#see signature of the download_house_list_data method from data_ingestion file
inspect.signature(data_ingestion.download_house_list_data)

<Signature (url)>

In [6]:
inspect.getdoc(data_ingestion.download_house_list_data)

'downloads dataset from kaggle data scourse by using url of the dataset file in website.\n\nArgs:\n    url (String): web site url for dataset.\n\nReturns:\n    Saves downloaded file in location.'

In [7]:
data_ingestion.download_house_list_data("https://www.kaggle.com/datasets/tahirmehmood0/lahore-house-listings-from-zameen-com-2025")

Dataset URL: https://www.kaggle.com/datasets/tahirmehmood0/lahore-house-listings-from-zameen-com-2025
dataset downloaded Successfully!
Filename: C:\Users\KISHORE\OneDrive\Desktop\Intelligent_Weather_Prediction_Pipeline\scripts\data\data_ingestion.py

Line #    Mem usage    Increment  Occurrences   Line Contents
    65    223.7 MiB    223.7 MiB           1   @profile
    66                                         def download_house_list_data(url):
    67                                             """
    68                                             downloads dataset from kaggle data scourse by using url of the dataset file in website.
    69                                         
    70                                             Args:
    71                                                 url (String): web site url for dataset.
    72                                         
    73                                             Returns:
    74                                         

- **for downloading dataset system took upto 5.3 increament in memory**

In [8]:
inspect.signature(data_ingestion.read_data)

<Signature ()>

In [9]:
inspect.getdoc(data_ingestion.read_data)

'Reads the downloaded dataset file from dataset location.\n\nArgs:\n    Nothing\n\nReturns:\n    Dataset(DataFrame): it contains Lahore House Listings from Zameen.com (2025).'

In [10]:
lohore_dataset = data_ingestion.read_data()

Filename: C:\Users\KISHORE\OneDrive\Desktop\Intelligent_Weather_Prediction_Pipeline\scripts\data\data_ingestion.py

Line #    Mem usage    Increment  Occurrences   Line Contents
    90    229.3 MiB    229.3 MiB           1   @profile
    91                                         def read_data():
    92                                             """
    93                                             Reads the downloaded dataset file from dataset location.
    94                                         
    95                                             Args:
    96                                                 Nothing
    97                                         
    98                                             Returns:
    99                                                 Dataset(DataFrame): it contains Lahore House Listings from Zameen.com (2025).
   100                                             """
   101    235.7 MiB      6.5 MiB           1       dataset = pd.read_csv(RO

In [11]:
type(lohore_dataset)

pandas.core.frame.DataFrame

### Data Inspection

In [12]:
lohore_dataset.head(2)

,Title,Location,Type,Price,Area,Bedrooms,Bathrooms,Purpose,Date Posted,Built Year,...,Furnished,Gym,Study Room,Drawing Room,Dining Room,Lawn/Garden,Swimming Pool,Electricity Backup,Lounge/Sitting Room,Link
0,ACTUALLY 25 MARLA DOUBLE UNIT LUXURIOUS & GRAC...,"DHA Phase 6 - Block F, DHA Phase 6, DHA Defenc...",House,8.15 Crore,1 Kanal,6,7,For Sale,2025-07-27,2024.0,...,True,True,True,True,True,True,True,True,True,https://www.zameen.com/Property/dha_phase_6_dh...
1,3 Years Installment Plan Luxury Brand New Hous...,"Park View City, Lahore, Punjab",House,1.5 Crore,5 Marla,4,5,For Sale,2025-07-25,2012.0,...,True,False,False,True,True,True,False,True,True,https://www.zameen.com/Property/lahore_park_vi...


In [13]:
lohore_dataset.sample(n=500).head(2)

,Title,Location,Type,Price,Area,Bedrooms,Bathrooms,Purpose,Date Posted,Built Year,...,Furnished,Gym,Study Room,Drawing Room,Dining Room,Lawn/Garden,Swimming Pool,Electricity Backup,Lounge/Sitting Room,Link
13144,5 Marla Beautiful House For Sale,"Al Kareem Garden, GT Road, Lahore, Punjab",House,1.35 Crore,5 Marla,4,5,For Sale,2025-07-27,2024.0,...,True,True,True,True,True,True,True,True,True,https://www.zameen.com/Property/gt_road_al_kar...
5170,1 Kanal Beautiful House For Sale,"DHA Phase 2, DHA Defence, Lahore, Punjab",House,6.15 Crore,1 Kanal,5,6,For Sale,NaN,2004.0,...,True,True,True,True,True,True,True,True,True,https://www.zameen.com/Property/dha_defence_dh...


In [14]:
lohore_dataset.tail(2)

,Title,Location,Type,Price,Area,Bedrooms,Bathrooms,Purpose,Date Posted,Built Year,...,Furnished,Gym,Study Room,Drawing Room,Dining Room,Lawn/Garden,Swimming Pool,Electricity Backup,Lounge/Sitting Room,Link
18247,3 Years Installment Plan Luxury Brand New Hous...,"Park View City, Lahore, Punjab",House,1.1 Crore,3.5 Marla,3,4,For Sale,2025-07-28,NaN,...,False,False,False,True,True,False,False,False,False,https://www.zameen.com/Property/lahore_park_vi...
18248,CUTTING-EDGE-DESIGN,"DHA Phase 6, DHA Defence, Lahore, Punjab",House,2.3 Crore,5 Marla,4,5,For Sale,2025-07-29,2024.0,...,False,False,False,False,False,False,False,True,False,https://www.zameen.com/Property/dha_defence_dh...


In [15]:
print("Shape of the dataset: ",lohore_dataset.shape)
print("Size of the dataset: ",lohore_dataset.size)

Shape of the dataset:  (18249, 23)
Size of the dataset:  419727


 **lohore dataset have total 23 atributes with 18249 Records.So dataset is high volume.**

In [16]:
lohore_dataset.columns

Index(['Title', 'Location', 'Type', 'Price', 'Area', 'Bedrooms', 'Bathrooms',
       'Purpose', 'Date Posted', 'Built Year', 'Kitchens', 'Store Rooms',
       'Servant Quarters', 'Furnished', 'Gym', 'Study Room', 'Drawing Room',
       'Dining Room', 'Lawn/Garden', 'Swimming Pool', 'Electricity Backup',
       'Lounge/Sitting Room', 'Link'],
      dtype='object')

In [17]:
d = {"Column": lohore_dataset.columns, "Dtype" : lohore_dataset.dtypes}
data_types_df = pd.DataFrame(d)
index = [""]*len(data_types_df)
data_types_df.index = index
data_types_df

,Column,Dtype
,Title,object
,Location,object
,Type,object
,Price,object
,Area,object
,Bedrooms,object
,Bathrooms,object
,Purpose,object
,Date Posted,object
,Built Year,float64


- **Dataset have object,int64,float64 as well as bool data types.**

### making sweetviz Report

In [18]:
os.chdir(ROOT_DIR/REPORTS)
start = time.time()
report = sv.analyze(lohore_dataset)
report.show_html("lohore_dataset_basic_report.html")
end = time.time()
print(f"TIme taken: {round(end-start,2)}sec")
cpu_percent = ps.cpu_percent(interval=1)
print(f"CPU Usage: {cpu_percent}%")
print("Report Downloaded Successfully!")

                                             |                                             | [  0%]   00:00 ->…

Report lohore_dataset_basic_report.html was generated! NOTEBOOK/COLAB USERS: the web browser MAY not pop up, regardless, the report IS saved in your notebook/colab files.
TIme taken: 31.43sec
CPU Usage: 55.9%
Report Downloaded Successfully!
